mne/time_frequency/tests/test_psd.py

```diff
@@ -13,102 +15,193 @@
+        n_fft = 2048  # the FFT size (n_fft). Ideally a power of 2
+        psds, freqs = compute_raw_psd(raw, tmin=tmin, tmax=tmax, picks=picks,
+                                      fmin=fmin, fmax=fmax, n_fft=n_fft,
+                                      n_jobs=1, proj=False)
+        psds_proj, freqs = compute_raw_psd(raw, tmin=tmin, tmax=tmax,
+                                           picks=picks, fmin=fmin, fmax=fmax,
+                                           n_fft=n_fft, n_jobs=1, proj=True)
 
-    assert_array_almost_equal(psds, psds_proj)
-    assert_true(psds.shape == (len(picks), len(freqs)))
-    assert_true(np.sum(freqs < 0) == 0)
-    assert_true(np.sum(psds < 0) == 0)
+        assert_array_almost_equal(psds, psds_proj)
+        assert_true(psds.shape == (len(picks), len(freqs)))
+        assert_true(np.sum(freqs < 0) == 0)
+        assert_true(np.sum(psds < 0) == 0)
 
 
 @requires_version('scipy', '0.12')
-def test_psd_epochs():
-    """Test PSD estimation on epochs
+def test_psd():
+    """Tests the welch and multitaper PSD
     """
     raw = io.Raw(raw_fname)
-
     exclude = raw.info['bads'] + ['MEG 2443', 'EEG 053']  # bads + 2 more
+    include = []
+
+    tmin, tmax = 0, 10  # use the first 60s of data
+    fmin, fmax = 2, 70  # look at frequencies between 5 and 70Hz
+    n_fft = 128
 
     # picks MEG gradiometers
-    picks = pick_types(raw.info, meg='mag', eeg=False, stim=False,
-                       exclude=exclude)
+    picks = pick_types(raw.info, meg='grad', eeg=False, eog=True,
+                       stim=False, include=include, exclude=exclude)
 
-    picks = picks[:2]
+    # -- Raw --
+    # Welch
+    kws_welch = dict(tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax, n_fft=n_fft)
+    psds, freqs = psd_welch(raw, proj=False, picks=picks, **kws_welch)
+    psds_proj, freqs_proj = psd_welch(raw, proj=True, picks=picks, **kws_welch)
 
-    n_fft = 512  # the FFT size (n_fft). Ideally a power of 2
+    assert_array_almost_equal(psds, psds_proj)
+    assert_true(psds.shape == (len(picks), len(freqs)))
+    assert_true(np.sum(freqs < 0) == 0)
+    assert_true(np.sum(psds < 0) == 0)
 
-    tmin, tmax, event_id = -0.5, 0.5, 1
-    include = []
-    raw.info['bads'] += ['MEG 2443']  # bads
+    # Multitaper
+    picks_mt = picks[:2]
+    kws_mt = dict(tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax, low_bias=True)
+    psds, freqs = psd_multitaper(raw, proj=False, picks=picks_mt, **kws_mt)
+    psds_proj, freqs_proj = psd_multitaper(raw, proj=True,
+                                           picks=picks_mt, **kws_mt)
 
-    # picks MEG gradiometers
-    picks = pick_types(raw.info, meg='grad', eeg=False, eog=True,
-                       stim=False, include=include, exclude='bads')
+    assert_array_almost_equal(psds, psds_proj)
+    assert_true(psds.shape == (len(picks_mt), len(freqs)))
+    assert_true(np.sum(freqs < 0) == 0)
+    assert_true(np.sum(psds < 0) == 0)
```

Comment:
yep I can do that - I thought that loops weren't recommended in test functions so that you always know which code broke the test, but I can loopify this for sure